.github/workflows/gurobi-auto-update.yml

```diff
@@ -0,0 +1,91 @@
+name: Gurobi Auto Updater
+
+on:
+  schedule:
+    - cron: '15 1 * * 3' # Every Wednesday at 1:15 AM UTC
+  workflow_dispatch:
+
+jobs:
+  update-gurobi-version:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Get Latest Gurobipy Version
+        id: get-latest-version
+        run: |
+          echo "NEW_VERSION=$(curl -s https://pypi.org/pypi/gurobipy/json | jq -r '.info.version')" >> $GITHUB_ENV
+
+      - name: Current Version
+        uses: zlatko-ms/varfiletoenv@v3
+        with:
+          paths: ./.github/gurobi_version.json
+
+      - name: Check if branch exists or version matches
+        id: check-branch-version
+        env:
+          NEW_VERSION: ${{ env.NEW_VERSION }}
+          GUROBI_VERSION: ${{ env.gurobiVersion }}
+        run: |
+          git fetch
+          echo "NEW_VERSION: $NEW_VERSION"
+          echo "GUROBI_VERSION: $GUROBI_VERSION"
+          if [[ $(git branch -a | grep "update-gurobi-$NEW_VERSION") ]]; then
+            echo "Branch update-gurobi-$NEW_VERSION already exists. Aborting workflow."
+            echo "CONTINUE_WORKFLOW=false" >> $GITHUB_ENV
+          elif [[ $NEW_VERSION == $GUROBI_VERSION ]]; then
+            echo "Gurobi version already included. Aborting workflow."
+            echo "CONTINUE_WORKFLOW=false" >> $GITHUB_ENV
+          else
+            echo "CONTINUE_WORKFLOW=true" >> $GITHUB_ENV
+          fi
+
+      - name: Create new branch
+        if: env.CONTINUE_WORKFLOW == 'true'
+        env:
+          NEW_VERSION: ${{ env.NEW_VERSION }}
+        run: |
+          git checkout -b update-gurobi-$NEW_VERSION
+          git push origin update-gurobi-$NEW_VERSION
```

Comment:
I'd probably combine all of these steps with the steps below to eliminate the duplication of quite some code. And all the conditional and environment variable code.
This should also help to simplify quite a couple of the git commands.

Reply:
True. I combined them in one step to reduce duplciate code. For some reason I still find creating the PR itself in a separate step sensible.